src/components/card/card.component.ts

```diff
@@ -0,0 +1,113 @@
+import { CustomElement, ICustomElement } from '../../lib/custom-element';
+
+import './card.component.scss';
+
+@CustomElement({
+  selector: 'app-card',
+  // template: `
+  //   <div class="app-card">
+  //     ${name}
+  //   </div>
+  // `,
+  // // Have in account: [:host] style definition only works when useShadow is set to true
+  // style: ``,
+  useShadow: true,
+})
+export class CardComponent extends HTMLElement implements ICustomElement {
+
+  templateElement: HTMLTemplateElement;
+  shRoot: ShadowRoot;
+  useShadow: boolean;
+
+  static get observedAttributes() {
+    return [
+      'character-name',
+    ];
+  }
+
+  constructor() {
+    super();
+    this.useShadow = true;
+    this.templateElement = document.createElement('template');
+    this.shRoot = this.attachShadow({ mode: 'open' });
+  }
+
+  attributeChangedCallback(name: string, oldValue, newValue): void {
+    // console.log('Attributes changed:', [name, oldValue, newValue]);
+    if (oldValue === newValue) {
+      return;
+    }
+    // console.log(`The attribute ${name} has changed`);
+  }
+
+  componentWillMount() {
+    // console.log('componentWillMount', this.getAttributeNames());
+  }
+
+  connectedCallback() {
+    // console.log('connectedCallback');
+    this.render();
+  }
+
+  disconnectedCallback() {
+    // console.log('disconnectedCallback');
+  }
+
+  componentDidMount() {
+    // console.log('componentDidMount');
+  }
+
+  componentWillUnmount() {
+    // console.log('componentWillUnmount');
+  }
+
+  componentDidUnmount() {
+    // console.log('componentDidUnmount');
+  }
+
+  render() {
+    const template = `
+      <style>
+        :host {
+          margin: 15px;
```

Comment:
Can you have all these styles in a separate file and import them here?

Reply:
I can't import it from outside because I need to inject the styles as string to Shadow DOM. In a next commit I'm injecting the styles from class decorator.